### DISCUSSION TITLE
all Destinations

### DISCUSSION THREAD

**Discussion by linlinsky2023** [pymupdf/PyMuPDF#2754] [General]
How I can get the all Destinations,like pypdf2 reader.getNamedDestinations()

**Comment by linlinsky2023** [comment]
I get the link dict like this {'kind': 4, 'xref': 50, 'from': Rect(141.74899291992188, 92.5460205078125, 199.34100341796875, 105.8489990234375), 'name': 'nameddest=F201.090000', 'id': '}. how I can get the page number by the "nameddest=F201.090000"

**Reply by JorjMcKie** [reply]
Try this new helper script: https://github.com/pymupdf/PyMuPDF-Utilities/tree/master/pdf-names-resolution

**Reply by linlinsky2023** [reply]
if obj_type == "xref":
        dests_xref = int(dests_xref.split()[0])
    else:
        return {}
but my obj_type == "dict" and return {}

**Reply by JorjMcKie** [reply]
let me have your PDF please

**Reply by JorjMcKie** [reply]
There is a second script in the said folder, which resolves the above problem.

**Comment by JorjMcKie** [comment]
@linlinsky2023 with the new version 1.23.6 you can do this:
```python
In [1]: import fitz_new as fitz
In [2]: doc=fitz.open("pymupdf.pdf")
In [3]: names = doc.resolve_names()
In [4]: names["Doc-Start"]
Out[4]: {'page': 0, 'to': (72.0, 720.0), 'zoom': 0.0}
In [5]: fitz.__version__
Out[5]: '1.23.6'
In [6]: doc=fitz.open("pandas.pdf")
In [7]: names = doc.resolve_names()
In [8]: names["Doc-Start"]
Out[8]: {'page': 0, 'to': (72.0, 720.0), 'zoom': 0.0}
In [9]: len(names)
Out[9]: 12912
In [10]: list(names.items())[5000]
Out[10]: ('paragraph*.5062', {'page': 2299, 'to': (72.0, 586.255), 'zoom': 0.0})
In [11]:
```

**Reply by JorjMcKie** [reply]
Note: this feature is supported as above only. The new "rebased" architecture will become the default PyMuPDF implementation soon. You can already use it today as shown as a plugin replacement.
